libraries/AP_HAL_ChibiOS/hwdef/AEROFOX-H7/hwdef.dat

```diff
@@ -0,0 +1,225 @@
+# MCU class and specific type
+MCU STM32H7xx STM32H743xx
+
+# crystal frequency
+OSCILLATOR_HZ 24000000
+
+# board ID for firmware load
+APJ_BOARD_ID 7200
+
+FLASH_SIZE_KB 2048
+
+# bootloader is installed at zero offset
+FLASH_RESERVE_START_KB 128
+
+# UART1 is debug
+PA9 USART1_TX USART1 NODMA
+PA10 USART1_RX USART1 NODMA
+
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1# crystal frequency
+
+# supports upto 8MBits/s
+CANFD_SUPPORTED 8
+
+# with 2M flash we can afford to optimize for speed
+env OPTIMIZE -O2
+
+# system timer
+STM32_ST_USE_TIMER 5
+
+# UART
+SERIAL_ORDER OTG1 UART7 UART4 UART5 USART2 USART1 UART8 USART3 OTG2
+
+#serial1
+PA15 UART7_TX UART7
+PA8 UART7_RX UART7
+
+# serial2
+PC10 UART4_TX UART4
+PC11 UART4_RX UART4
+
+# serial3
+PC12 UART5_TX UART5
+PD2 UART5_RX UART5
+
+# serial4
+PD5 USART2_TX USART2
+PD6 USART2_RX USART2
+
+# serial6
+PE1 UART8_TX UART8
+PE0 UART8_RX UART8
+
+# serial7
+PD8 USART3_TX USART3
+PD9 USART3_RX USART3
+
+# default the 2nd interface to SLCAN
+define HAL_OTG2_PROTOCOL SerialProtocol_SLCAN
+
+# CAN
+PD0  CAN1_RX CAN1
+PD1  CAN1_TX CAN1
+
+PB12 CAN2_RX CAN2
+PB13 CAN2_TX CAN2
+
+# I2C
+I2C_ORDER I2C1 I2C2
+
+PB6  I2C1_SCL I2C1
+PB7  I2C1_SDA I2C1
+
+PB10 I2C2_SCL I2C2
+PB11 I2C2_SDA I2C2
+
+# SPI
+PB3 SPI1_SCK   SPI1
+PB4 SPI1_MISO  SPI1
+PB5 SPI1_MOSI  SPI1
+
+PD3 SPI2_SCK  SPI2
+PC2 SPI2_MISO SPI2
+PC3 SPI2_MOSI SPI2
+
+PE2 SPI4_SCK   SPI4
+PE5 SPI4_MISO  SPI4
+PE6 SPI4_MOSI  SPI4
+
+# IMU
+IMU ADIS1647x    SPI:16470    ROTATION_PITCH_180 ADIS_DRDY_PIN
+IMU Invensensev3 SPI:42688_1  ROTATION_NONE
+IMU Invensensev3 SPI:42688_2  ROTATION_PITCH_180_YAW_270
+
+SPIDEV 16470           SPI1 DEVID1  16470_CS        MODE3  1*MHZ  2*MHZ
+SPIDEV 42688_1        SPI1 DEVID2  42688_1_CS     MODE3  2*MHZ  8*MHZ
+SPIDEV 42688_2        SPI4 DEVID3  42688_2_CS     MODE3  2*MHZ  8*MHZ
+
+SPIDEV ramtron        SPI2 DEVID1  FRAM_CS        MODE3  8*MHZ  8*MHZ
+SPIDEV sdcard          SPI2 DEVID2  FLASH_CS       MODE0   1*MHZ    8*MHZ
```

Comment:
Ideally you would use SDMMC here

Reply:
Thank you for your advice, but we don't have enough GPIO ports on our hardware to use SDMMC